presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -1088,4 +1088,18 @@ void setS3Client(AmazonS3 client)
     {
         s3 = client;
     }
+
+    /**
+     * Helper function used to work around the fact that if you use an S3 bucket with an '_' that java.net.URI
+     * does behaves differently and sets the host value to null where as S3 buckets without '_' have a properly
+     * set host field.
+     *
+     * @param uri The URI from which to extract a host value.
+     * @return The host value where uri.getAuthority() is used when uri.getHost() returns null.
+     * @note '_' is only allowed in S3 bucket names in us-east-1
+     */
+    String safeGetHost(URI uri)
```

Comment:
Let's name this `getBucketName()`, since we are using this to extract the bucket name from the URI to pass to various APIs.

Reply:
I like this, will make that change.